vulkan/vulkan_device.h

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_VULKAN_VULKAN_DEVICE_H_
+#define FLUTTER_VULKAN_VULKAN_DEVICE_H_
+
+#include <vector>
+#include "flutter/vulkan/vulkan_handle.h"
+#include "lib/ftl/compiler_specific.h"
+#include "lib/ftl/macros.h"
+
+namespace vulkan {
+
+class VulkanProcTable;
+class VulkanSurface;
+
+class VulkanDevice {
+ public:
+  VulkanDevice(VulkanProcTable& vk,
+               VulkanHandle<VkPhysicalDevice> physical_device);
+
+  ~VulkanDevice();
+
+  bool IsValid() const;
+
+  const VulkanHandle<VkDevice>& Handle() const;
+
+  const VulkanHandle<VkPhysicalDevice>& PhysicalDeviceHandle() const;
```

Comment:
Not sure about Flutter style... should these be GetPhysicalDeviceHandle(), etc. ?  Or physical_device_handle() ?